src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DefaultLSPProgressListener.cs

```diff
@@ -40,13 +41,15 @@ public DefaultLSPProgressListener(ILanguageServiceBroker2 languageServiceBroker)
             string token,
             Func<JToken, CancellationToken, Task> onProgressNotifyAsync,
             TimeSpan timeoutAfterLastNotify,
+            CancellationToken immediateNotificationTimeout,
```

Comment:
Are there any product benefits for having this exposed in the normal pipeline? Would it make more sense to have a test-only property that can set this and have the product code react to it being non-null?

Reply:
It's not clear to me how the value would get passed through otherwise.

Reply:
Ya I was suggesting changing the scoping to be at the DefaultLSPProgressListener level to basically have a "bail if test" type functionality. That being said, if our tests expect multiple of these request scoped listeners to be live simultaneously with one of them timing out then of course that conflicts with the suggestion.

Reply:
@NTaylorMullen is this something you want me to look at more or use like this for now?

Reply:
If it's quick and easy to move to a test only property then I'd say do it now. If it's a little more involved I'd say leave as is.

Reply:
Can you check the update?

Reply:
Really like what you came up with. 👏